Tests/AlicerceTests/Analytics/AnalyticsTestCase.swift

```diff
@@ -0,0 +1,154 @@
+        let analyticsConfiguration = Analytics.Configuration(extraParameters: [
+            "1" : "1"
+        ])
+        
+        let analytics = Analytics(configuration: analyticsConfiguration)
+        
+        XCTAssertNotNil(analytics.parameters)
+        XCTAssertEqual(analytics.parameters?.count, 1)
+        
+        analytics.add(parameters: ["2" : "2"])
+        
+        XCTAssertEqual(analytics.parameters?.count, 2)
+        XCTAssertEqual(analytics.parameters!["1"] as! String, "1")
+        XCTAssertEqual(analytics.parameters!["2"] as! String, "2")
+    }
+    
+    func testAddTracker_WhenATrackerIsAdded_ItShouldSendEventsToThatTracker() {
+        let expectation = self.expectation(description: "Track appended tracker")
+
+        let page = Analytics.Page(name: "trackPage", parameters: nil)
+        let testableTracker = TestableTracker(expectation: expectation, page: page)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: testableTracker)
+        
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackPage_WithoutGlobalParameters_ItShouldMatchTheOriginalPage() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let page = Analytics.Page(name: "trackablePage", parameters: ["1" : "1"])
+        let tracker = TestableTracker(expectation: expectation, page: page)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: tracker)
+        
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackPage_WithGlobalParameters_ItShouldMatchAPageWithMoreParameters() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let trackerPage = Analytics.Page(name: "trackablePage", parameters: ["1" : "1", "2" : "2"])
+        let tracker = TestableTracker(expectation: expectation, page: trackerPage)
+        
+        let analyticsConfiguration = Analytics.Configuration(extraParameters: [
+            "1" : "1"
+            ])
+        
+        let analytics = Analytics(configuration: analyticsConfiguration)
+        analytics.add(tracker: tracker)
+        
+        let page = Analytics.Page(name: "trackablePage", parameters: ["2" : "2"])
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackEvent_WithoutGlobalParameters_ItShouldMatchOriginalEvent() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let event = Analytics.Event(name: "trackableEvent", parameters: ["1" : "1"])
+        let tracker = TestableTracker(expectation: expectation, event: event)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: tracker)
+        
+        analytics.track(event: event)
+        
```

Comment:
should we take that to that level? it looks for me like we are testing not this functionality from others components, not this one.